packages/service-library/src/servicelib/async_utils.py

```diff
@@ -157,13 +174,18 @@ async def worker(in_q: Queue, out_q: Queue) -> None:
                         target_args,
                     )
 
-                context.task = asyncio.create_task(
-                    worker(context.in_queue, context.out_queue)
-                )
+                with request_profiler():
+                    context.task = asyncio.create_task(
+                        worker(context.in_queue, context.out_queue)
+                    )
 
-            await context.in_queue.put(decorated_function(*args, **kwargs))
+            with request_profiler() as profiler:
```

Comment:
here maybe if you can add somewhere a NOTE on why you do that?
minor: ideally a test would have nice too.

Reply:
done, thanks!